src/utils/transaction.ts

```diff
@@ -314,6 +421,308 @@ export const buildCrossChainAction = async (
     }
   }
 
+  if (transactionBlock.type === TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE
+    && !!transactionBlock?.values?.fromChainId
+    && !!transactionBlock?.values?.fromAsset
+    && !!transactionBlock?.values?.amount
+    && !!transactionBlock?.values?.offer
+    && !!transactionBlock?.values?.receiverAddress
+    && !!transactionBlock?.values?.toAsset) {
+    if(!transactionBlock?.values?.hasEnoughPLR){
```

Comment:
combine under single please
```suggestion
    && !!transactionBlock?.values?.toAsset) {
    if(!transactionBlock?.values?.hasEnoughPLR){
```

Reply:
We need if condition (`if(!transactionBlock?.values?.hasEnoughPLR)`) to handle swap and staking flow.